stl/inc/span

```diff
@@ -226,28 +226,30 @@ struct pointer_traits<_Span_iterator<_Ty>> {
 };
 
 // STRUCT TEMPLATE _Span_extent_type
-template <size_t _Extent>
+template <class _Ty, size_t _Extent>
 struct _Span_extent_type {

```

Comment:
As a general rule, we try to use `struct` for things that have entirely `public` members, and `class` for things that keep secrets. (We've accumulated some exceptions due to maintenance, or the Standard specifying specifically `struct` or `class` combined with MSVC compiler bugs regarding switching between the two.)

Here and in the partial specialization, `struct _Span_extent_type` has `protected` data members, so it doesn't follow the ideal convention. It essentially keeps no secrets from `span`. Now that `span` inherits privately from `_Span_extent_type`, I think there's a strong case for simply removing `protected` here (and the partial specialization) and making this a true `struct`.

Reply:
:+1:

I thought about that but ended up being lazy